src/hotspot/cpu/x86/gc/g1/g1BarrierSetAssembler_x86.cpp

```diff
@@ -314,18 +317,13 @@ void G1BarrierSetAssembler::g1_write_barrier_post(MacroAssembler* masm,
 
   __ movb(Address(card_addr, 0), (int)G1CardTable::dirty_card_val());
 
-  __ cmpl(queue_index, 0);
+  __ movptr(tmp2, queue_index);
+  __ testptr(tmp2, tmp2);
   __ jcc(Assembler::equal, runtime);
```

Comment:
I know it is the same in x86-speak, but `Assembler::zero` would be more readable to show that we actually test `queue_index == 0`.

Reply:
You mean using `__ cmpptr(tmp2, 0)` or something similar? I would like to keep this that way, even if it's only a nano-optimization.

Otherwise we should also change the pre-barrier code to be uniform.

Reply:
No, I mean `testptr(tmp2, tmp2); jcc(Assembler::zero, runtime)`. In fact, I see the similar code on L489 in the same file.